src/app/chart-index/chart-index.component.ts

```diff
@@ -28,6 +30,23 @@ export class ChartIndexComponent implements OnInit {
       this.loading = false;
       this.charts = charts;
       this.totalChartsNumber = charts.length;
+    }, () => {
+      this.loading = false;
+      this.apiNotReady = true;
+      IntervalObservable.create(2000)
+        .takeWhile(() => this.apiNotReady)
```

Comment:
I guess the `takeWhile` will keep the observer when the `this.apiNotReady` value is `true`. However, I'd add a more explicit check to avoid wrong values in this variable.

```
.takeWhile(() => this.apiNotReady === true)
```